### PR TITLE
Add KernelAbstractions.jl support

### PR DESCRIPTION
**Pull request by michel2323**
This makes ExaTron support everything that is needed for KA support in ExaAdmm.jl https://github.com/exanauts/ExaAdmm.jl/issues/33

### CODE REVIEW

**Review by frapac — COMMENTED** [review]
That's a great work overall, and a step forward. I am glad KA is coming to ExaTron. 

This PR is quite long, and my review is far from being exhaustive. As a general comment, I am wondering why we pass the thread number as an argument now? 

Also, ExaTron builds a single kernel by using extensively `@inline` instruction and nesting functions together. As a general feedback, why is your experience with that? Have you encountered some issues with KA?

File: src/CUDA/TronMatrix.jl [hunk 1]
```diff
@@ -0,0 +1,24 @@
+function ExaTron.TronDenseMatrix(I::VI, J::VI, V::CuArray, n) where {VI}
```
[the remark this answers is not in the document]
Comment (michel2323) [reply src/CUDA/TronMatrix.jl:1]: Nice!

File: src/CUDA/TronMatrix.jl [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    A = TronDenseMatrix{CuArray{Float64, 2}}(n, n, tron_zeros(CuArray{eltype(V)}, (n, n)))
+    for i=1:length(I)
+        @assert 1 <= I[i] <= n && 1 <= J[i] <= n && I[i] >= J[i]
+        @inbounds A.vals[I[i], J[i]] += V[i]
+    end
+
+    return A
+end
+
+@inline function ExaTron.Base.fill!(w::CuDeviceArray{Float64,1}, val::Float64)
```
[the remark this answers is not in the document]
Comment (michel2323) [reply src/CUDA/TronMatrix.jl:14]: Some `sed` mishap :-).

File: test/KA.jl [hunk 3]
```diff
@@ -0,0 +1,1402 @@
...
+
+@testset "dbreakpt" begin
+    @kernel function dbreakpt_test(::Val{n},dx,
+                            dl,
+                            du,
+                            dw,
+                            d_nbrpt,
+                            d_brptmin,
+                            d_brptmax
+                            ) where {n}
```
[the remark this answers is not in the document]
Comment (michel2323) [reply test/KA.jl:622]: `@localmem` doesn't support dynamic sizes.

**Pull request merged**